src/Fixer/ReturnNotation/ReturnAssignmentFixer.php

```diff
@@ -121,13 +121,16 @@ private function fixFunction(Tokens $tokens, int $functionIndex, int $functionOp
             T_INCLUDE_ONCE,               // "
             T_REQUIRE,                    // "
             T_REQUIRE_ONCE,               // "
-            T_STATIC,
         ];
 
         $inserted = 0;
         $candidates = [];
         $isRisky = false;
 
+        if ($tokens[$tokens->getNextMeaningfulToken($functionIndex)]->isGivenKind(CT::T_RETURN_REF)) {
```

Comment:
I think this cannot happen because of the [same check in applyFix()](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/f403e7f9108add56c4d3f0c0e3f0e13845737e74/src/Fixer/ReturnNotation/ReturnAssignmentFixer.php#L77=).

Reply:
The proof is really in the pudding :)
If you check out this branch and set a breakpoint on the line below (131) and run the tests in
`tests/Fixer/ReturnNotation/ReturnAssignmentFixerTest.php` you'll see it will hit.

The thing is within `applyFix()` there is a recursive call to simplify the returns, fixing a function does not only happen there, nested function fixing happen here.